File: queue/python/queue.py

class Queue:
    def __init__(self):
        self.items = []

    # dequeue - O(1)
    def dequeue(self):
        return self.items.pop()

    # enqueue - O(n)
    def enqueue(self, item):
        self.items.insert(0, item)

    # size
    def size(self):
        return len(self.items)

    # is empty
    def is_empty(self):
        return self.items == []

File: queues_and_stacks/classes/queue/queue.js

// We almost always can just use a normal JS array to implement this
// without writing a class like the following.

class Queue {

  constructor() {
    this.list = [];
    this.length = 0;
  }

  enqueue(value) {
    this.length++;
    this.list.push(value);
  }

  dequeue() {

    // Don't do anything if we don't have any items.
    if (this.length === 0) return;

    // Shift the first item off the start of the list and return the value.
    this.length--;
    return this.list.shift();
  }

  peek() {
    return this.list[0];
  }
}
